Qq/Commands.lean

```diff
@@ -0,0 +1,97 @@
+import Qq.Macro
+import Lean
+
+namespace Qq
+open Lean Meta Elab Tactic
+
+private
+def mkHaveFVars (assignments : Array Expr) (body : Expr) : MetaM Expr := do
+  let ctx ← getLCtx
+  let (_, res) ← ctx.foldrM (init := (assignments, body)) fun decl (assignments, body) => do
+    let value := assignments.back!
+    let result := .letE decl.userName decl.type value (body.abstract #[.fvar decl.fvarId]) false
+    return (assignments.pop, result)
+  return res
+
+/--
+`by_elabq` is the Qq analogue to `by_elab`
+Executes arbitrary `TermElabM` code in place of a term.
+Moreover, all variables available in the local context are accessible as
+quoted expressions, and the return value is Q-annotated as well.
+Example:
+```
+def f (x : Prop) [Decidable x] : Int :=
+  by_elabq
+    Lean.logInfo x
+    Lean.logInfo x.ty
+    return q(if $x then 2 else 3)
```

Comment:
Could you create a test file with this and more complex examples?

Reply:
in `examples`? ok, I will look at it

Reply:
Added examples/commandShowcase.lean